optimism/src/keccak/constraints.rs

```diff
@@ -26,8 +30,38 @@ impl<Fp: Field> Constraints for KeccakEnv<Fp> {
     }
 
     fn constraints(&mut self) {
-        todo!();
         // CORRECTNESS OF FLAGS
+        {
+            // TODO: remove redundancy if any
+
+            // Booleanity of sponge flags
+            {
+                // Absorb is either true or false
+                self.constrain(Self::boolean(self.absorb()));
+                // Squeeze is either true or false
+                self.constrain(Self::boolean(self.squeeze()));
```

Comment:
Nit: I would have called the method `is_boolean`. Not part of this PR.

Reply:
oh, yeah I can modify this. I just inherited a function we had in kimchi that was called just `boolean`